crates/ruff_python_parser/src/parser/statement.rs

```diff
@@ -3160,24 +3160,94 @@ impl<'src> Parser<'src> {
         // We should do the same but currently we can't without throwing away the parsed
         // expression because the AST can't contain it.
 
+        // test_ok type_param_type_var_tuple
+        // type X[*Ts] = int
+        // type X[*Ts = int] = int
+        // type X[*Ts = *int] = int
+        // type X[T, *Ts] = int
+        // type X[T, *Ts = int] = int
         if self.eat(TokenKind::Star) {
             let name = self.parse_identifier();
 
+            let default_value = if self.eat(TokenKind::Equal) {
+                if self.at_expr() {
+                    // test_err type_param_type_var_tuple_invalid_default_expr
+                    // type X[*Ts = *int] = int
```

Comment:
Can you add one more test case to check the starred expression precedence (`bitwise_or`)? This could be:

```python
type X[*Ts = *int or str] = int
```

Reply:
Sure. For reference here's what CPython does in my implementation:

```
>>> type X[*Ts = *int or str] = int
  File "<stdin>", line 1
    type X[*Ts = *int or str] = int
                      ^^
SyntaxError: invalid syntax
```

And Ruff produces:
```
+2 | type X[*Ts = *int or str] = int
+  |               ^^^^^^^^^^ Syntax Error: Boolean expression cannot be used here
```

Reply:
Thank you! This is what I expected